discord/commands/options.py

```diff
@@ -141,10 +141,10 @@ class Option:
     max_length: Optional[:class:`int`]
         The maximum length of the string that can be entered. Must be between 1 and 6000 (inclusive).
         Only applies to Options with an :attr:`input_type` of :class:`str`.
-    autocomplete: Optional[:class:`Any`]
-        The autocomplete handler for the option. Accepts an iterable of :class:`str` or :class:`OptionChoice`, a callable (sync or async)
-        that takes a single argument of :class:`AutocompleteContext`, or a coroutine.
-        Must resolve to an iterable of :class:`str` or :class:`OptionChoice`.
+    autocomplete: Optional[Callable[[:class:`.AutocompleteContext`], Awaitable[Union[Iterable[:class:`.OptionChoice`], Iterable[:class:`str`], Iterable[:class:`int`], Iterable[:class:`float`]]]]]
+        The autocomplete handler for the option. Accepts a callable (sync or async)
+        that takes a single argument of :class:`AutocompleteContext`.
+        :func:`discord.utils.basic_autocomplete` may be used to create the callable.
```

Comment:
Ok so indeed, the original was wrong but i get what it was trying to convey: the callable must ***return*** a list of `str` or `OptionChoice`. I recokn the following language would be more conclusive:
```suggestion
        The callable must return an iterable of :class:`str` or :class:`OptionChoice`.
        Alternatively, :func:`discord.utils.basic_autocomplete` may be used in place of the callable.
```

Reply:
Oh awesome! I think that'd work, but I'll quickly check that on my end as well.

Reply:
Not sure it's as cut and dry as returning an interable string.
![image](https://github.com/Pycord-Development/pycord/assets/86581171/da31a454-5fd7-4c83-a528-9bb3be241ec1)
This produces the error of 
`TypeError: boing() takes 0 positional arguments but 1 was given`

I'm gonna try to see what it's passing through

Reply:
Never mind. Just having 1 singular parameter fixed it for some reason. Super odd, but it works.

Reply:
Seems like it prints out an autocomplete context... I guess that could be super useful if used right!

Reply:
Oh lmao.... you already have that parameter thing listed